metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1891,6 +1894,46 @@ class MetalsLanguageServer(
         .orNull
     }.asJava
 
+  @JsonRequest("metals/findTextInDependencyJars")
+  def findTextInDependencyJars(
+      params: FindTextInDependencyJarsRequest
+  ): CompletableFuture[util.List[Location]] = {
+    def readMask: Future[String] = Option(params.mask) match {
+      case Some(mask) =>
+        Future.successful(mask)
+      case None =>
+        languageClient
+          .metalsInputBox(
+            MetalsInputBoxParams(value = ".conf", prompt = "Enter file mask")
+          )
+          .asScala
+          .map(result => result.value)
+    }
+
+    def readContent: Future[String] = Option(params.content) match {
+      case Some(content) =>
+        Future.successful(content)
+      case None =>
+        languageClient
+          .metalsInputBox(
+            MetalsInputBoxParams(
+              prompt = "Enter content to search for"
+            )
+          )
+          .asScala
+          .map(result => result.value)
```

Comment:
```suggestion          
          .map{
            case name if !name.cancelled && name.value.nonEmpty =>
               Some(name.value)
            case _ =>
               None
          }
```

Or you could also use a function like:
```scala
def checkResult(result: MetalsInputBoxResult) = result match {
  case name if !name.cancelled && name.value.nonEmpty =>
      Some(name.value)
  case _ =>
      None
}
```

and use it for both params.  You will not have to check for `mask == null || content == null)` you will just be able to pass an Option. I think especially `mask` can be empty here.